src/_gettsim/functions/loader.py

```diff
@@ -26,56 +33,188 @@ def load_functions_for_date(date: datetime.date) -> list[PolicyFunction]:
     functions:
         The policy functions that are active at the given date.
     """
-    return [f for f in _load_internal_functions() if f.is_active_at_date(date)]
+    return _build_functions_tree_for_date(
+        date=date, package_root=RESOURCE_DIR, roots=PATHS_TO_INTERNAL_FUNCTIONS
+    )
 
 
-def _load_internal_functions() -> list[PolicyFunction]:
+def _build_functions_tree_for_date(
```

Comment:
Remove lines 23--43. Given that this function is private, there is no need to differentiate (else one might have thought of loading from a different location). In any case, naming is confusing as it stands, so let's just simplify...